server/odc-service/src/main/java/com/oceanbase/odc/service/resource/k8s/builder/NativeK8sConfigMapOperatorBuilder.java

```diff
@@ -53,7 +57,15 @@ protected boolean doMatch(String type) {
 
     @Override
     public K8sConfigMap toResource(ResourceEntity e, Optional<K8sConfigMap> runtimeResource) {
-        K8sConfigMap k8sConfigMap = new K8sConfigMap(new ResourceLocation(e), e.getStatus());
+        ResourceState nextState = e.getStatus();
+        try {
+            K8sConfigMapStatusDfa dfa = K8sConfigMapStatusDfa.getInstance();
+            dfa.setCurrentState(e.getStatus());
```

Comment:
If DFA is a static instance, this set and compute is not thread safe cause i's not automatic operation. why not set a function like next(State currentState, Option<T> resource) to determinate next state.

Reply:
no, dfa is not a static instance, you. may check this static method, we return a new object in it~

Reply:
I have changed the dfa's def and change the method name from `getInstance` to `buildInstance`